etl/steps/data/garden/artificial_intelligence/2023-06-21/epoch_aggregates.py

```diff
@@ -0,0 +1,53 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import cast
+
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("epoch_aggregates.start")
+
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = cast(Dataset, paths.load_dependency("epoch"))
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["epoch"]
+    df = pd.DataFrame(tb)
+    df["publication_date"] = pd.to_datetime(df["publication_date"])
+    df["year"] = df["publication_date"].dt.year
+    domain_counts = df.groupby(["year", "domain"]).size().reset_index(name="count")
+    organization_categorization_counts = (
+        df.groupby(["year", "organization_categorization"]).size().reset_index(name="count")
+    )
+    df_pivot_domain = domain_counts.pivot(index="year", columns="domain", values="count")
+    df_pivot_org = organization_categorization_counts.pivot(
+        index="year", columns="organization_categorization", values="count"
+    )
```

Comment:
I'd structure this differently and add some comments, so that it is easier to follow:

```python
tb = pd.DataFrame(ds_meadow["epoch"])
# Get year and date
df["publication_date"] = pd.to_datetime(df["publication_date"])
df["year"] = df["publication_date"].dt.year
# Get domain counts
domain_counts = df.groupby(["year", "domain"]).size().reset_index(name="count")
df_pivot_domain = domain_counts.pivot(index="year", columns="domain", values="count")
# Get organization categorisation counts
organization_categorization_counts = (
     df.groupby(["year", "organization_categorization"]).size().reset_index(name="count")
)
df_pivot_org = organization_categorization_counts.pivot(
    index="year", columns="organization_categorization", values="count"
)
```